client/src/main/scala/client/BackPressuredWebSocketClient.scala

```diff
@@ -63,13 +72,13 @@ final class BackPressuredWebSocketClient private (url: String)
     }
     catch {
       case ex: Throwable =>
-        Observable.error(ex)
+        Observable.raiseError(ex)
     }
 
-    val source = channel.timeout(5.seconds)
-      .doOnCanceled(closeConnection(webSocket))
+    val source = channel.dropByTimespan(1.second)
+      .doOnCancel(closeConnection(webSocket))
```

Comment:
1. I do not understand why you did `dropByTimespan`. The original code was saying that if the source goes silent for `5.seconds`, then consider that to be a timeout and reconnect.
2. `doOnCancel` no longer means what you think it means. The old behavior is now in `Observable.doOnDownstreamStop` ([see api](https://goo.gl/PoCRRH)). It doesn't matter, this call is not needed anymore, because we are going to move this logic in `createChannel`.